experiments/kompanion/cmd/export/export.go

```diff
@@ -59,59 +59,23 @@ const (
 )
 
 func BuildExportCmd() *cobra.Command {
-	var opts ExportOptions
-
+	opts := NewExportOptions()
 	cmd := &cobra.Command{
 		Use:     "export",
 		Short:   "export Config Connector resources",
 		Example: examples,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return RunExport(cmd.Context(), &opts)
+			return RunExport(cmd.Context(), opts)
 		},
 		Args: cobra.ExactArgs(0),
 	}
 
-	cmd.Flags().StringVarP(&opts.kubeconfig, kubeconfigFlag, "", opts.kubeconfig, "path to the kubeconfig file.")
-	cmd.Flags().StringVarP(&opts.reportNamePrefix, reportNamePrefixFlag, "", "report", "Prefix for the report name. The tool appends a timestamp to this in the format \"YYYYMMDD-HHMMSS.milliseconds\".")
-
-	cmd.Flags().StringArrayVarP(&opts.targetNamespaces, targetNamespacesFlag, "", []string{}, "namespace prefix to target the export tool. Targets all if empty. Can be specified multiple times.")
-	cmd.Flags().StringArrayVarP(&opts.ignoreNamespaces, ignoreNamespacesFlag, "", []string{"kube"}, "namespace prefix to ignore. Excludes nothing if empty. Can be specified multiple times. Defaults to \"kube\".")
-
-	cmd.Flags().StringArrayVarP(&opts.targetObjects, targetObjectsFlag, "", []string{}, "object name prefix to target. Targets all if empty. Can be specified multiple times.")
-	cmd.Flags().StringArrayVarP(&opts.ignoreObjects, ignoreObjectsFlag, "", []string{}, "object name prefix to ignore. Excludes nothing if empty. Can be specified multiple times.")
-
-	cmd.Flags().IntVarP(&opts.workerRountines, workerRoutinesFlag, "", 10, "Configure the number of worker routines to export namespaces with. Defaults to 10. ")
+	flags := cmd.Flags()
+	opts.AddFlags(flags)
 
 	return cmd
 }
 
-const (
-	// flag names.
-	kubeconfigFlag       = "kubeconfig"
-	reportNamePrefixFlag = "report-prefix"
-
-	targetNamespacesFlag = "target-namespaces"
-	ignoreNamespacesFlag = "exclude-namespaces"
-
-	targetObjectsFlag = "target-objects"
-	ignoreObjectsFlag = "exclude-objects"
-
-	workerRoutinesFlag = "worker-routines"
-)
-
-type ExportOptions struct {
-	kubeconfig       string
```

Comment:
Nit: typically we export these, but no objection to exporting only when we need them

Reply:
Made them public